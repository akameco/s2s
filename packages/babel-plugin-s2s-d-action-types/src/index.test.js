// @flow
import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

const filename = path.resolve(process.cwd(), 'app', 'counter', 'actions.js')

pluginTester({
  title: 'default',
  plugin,
  snapshot: true,
  babelOptions: { filename },
  tests: [
    `export type Action = Increment`,
    `
    // @flow
    export type Action = Increment
    `,
    `export type Action = Increment | Decrement`,
    {
      title: 'imports',
      code: `
// @flow
import type { X } from './types';

export type Action = Increment
  `,
    },
    {
      title: 'payload',
      code: `
export const INCREMENT: 'app/counter/Increment' = 'app/counter/Increment';

export type Increment = { type: typeof INCREMENT, payload: number };

export type Action = Increment;
  `,
    },
    {
      title: 'remove unuded action',
      code: `
export const INCREMENT: 'app/counter/Increment' = 'app/counter/Increment';

export type Increment = { type: typeof INCREMENT, payload: number };

export type Action = Decrement;
    `,
    },
    {
      title: 'snake_case',
      code: `export type Action = ThisIsAction;`,
    },
    {
      title: 'import type',
      code: `
import type { UserID } from './types.js'

export type FetchUser = {
  type: typeof FETCH_USER,
  id: UserID
};

export type Action = FetchUser;
      `,
    },
    {
      title: 'already has Actions',
      code: `
const x = 'x'

export const Actions = {}

export type Action = Add;
        `,
    },
    {
      title: 'Request/Failure/Success pattern',
      code: `
export type Action = FetchRequest;
        `,
    },
    {
      title: 'Request/Failure/Success pattern when Request/Success',
      code: `
export type Action = FetchRequest | FetchSuccess;
        `,
    },
    {
      title: 'Request/Failure/Success pattern when exist Success/Failure',
      code: `
export type Action = FetchSuccess | FetchFailure;
        `,
    },
    {
      title: 'no type Action',
      code: `// @flow`,
    },
    {
      title: 'no GenericTypeAnnotation',
      code: `export type Action = string`,
    },
    {
      title: 'default',
      code: `
// @flow
import * as constants from './constants'

export type Action = Increment
    `,
    },
  ],
})
