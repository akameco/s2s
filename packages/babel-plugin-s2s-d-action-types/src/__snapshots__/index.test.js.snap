// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default 1. s2s-d-action-types: 1. s2s-d-action-types 1`] = `
"
export type Action = Increment

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow

export type Increment = {
  type: typeof constants.INCREMENT
};

export type Action = Increment;
"
`;

exports[`default 2. s2s-d-action-types: 2. s2s-d-action-types 1`] = `
"
// @flow
export type Action = Increment

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow

export type Increment = {
  type: typeof constants.INCREMENT
};

export type Action = Increment;
"
`;

exports[`default 3. s2s-d-action-types: 3. s2s-d-action-types 1`] = `
"
export type Action = Increment | Decrement

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow

export type Increment = {
  type: typeof constants.INCREMENT
};
export type Decrement = {
  type: typeof constants.DECREMENT
};

export type Action = Increment | Decrement;
"
`;

exports[`default Request/Failure/Success pattern when Request/Success: Request/Failure/Success pattern when Request/Success 1`] = `
"
export type Action = FetchRequest | FetchSuccess;

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow

export type FetchRequest = {
  type: typeof constants.FETCH_REQUEST
};
export type FetchSuccess = {
  type: typeof constants.FETCH_SUCCESS
};
export type FetchFailure = {
  type: typeof constants.FETCH_FAILURE
};

export type Action = FetchRequest | FetchSuccess | FetchFailure;
"
`;

exports[`default Request/Failure/Success pattern when exist Success/Failure: Request/Failure/Success pattern when exist Success/Failure 1`] = `
"
export type Action = FetchSuccess | FetchFailure;

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow

export type FetchSuccess = {
  type: typeof constants.FETCH_SUCCESS
};
export type FetchFailure = {
  type: typeof constants.FETCH_FAILURE
};

export type Action = FetchSuccess | FetchFailure;
"
`;

exports[`default Request/Failure/Success pattern: Request/Failure/Success pattern 1`] = `
"
export type Action = FetchRequest;

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow

export type FetchRequest = {
  type: typeof constants.FETCH_REQUEST
};
export type FetchSuccess = {
  type: typeof constants.FETCH_SUCCESS
};
export type FetchFailure = {
  type: typeof constants.FETCH_FAILURE
};

export type Action = FetchRequest | FetchSuccess | FetchFailure;
"
`;

exports[`default already has Actions: already has Actions 1`] = `
"
const x = 'x'

export const Actions = {}

export type Action = Add;

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow

export type Add = {
  type: typeof constants.ADD
};

export type Action = Add;
"
`;

exports[`default default: default 1`] = `
"
// @flow
import * as constants from './constants.js'

export type Action = Increment

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import * as constants from './constants.js';

export type Increment = {
  type: typeof constants.INCREMENT
};

export type Action = Increment;
"
`;

exports[`default import type: import type 1`] = `
"
import type { UserID } from './types.js'

export type FetchUser = {
  type: typeof FETCH_USER,
  id: UserID
};

export type Action = FetchUser;

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import type { UserID } from './types.js';

export type FetchUser = {
  type: typeof FETCH_USER;
  id: UserID;
};

export type Action = FetchUser;
"
`;

exports[`default imports: imports 1`] = `
"
// @flow
import type { X } from './types';

export type Action = Increment

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import type { X } from './types';

export type Increment = {
  type: typeof constants.INCREMENT
};

export type Action = Increment;
"
`;

exports[`default no GenericTypeAnnotation: no GenericTypeAnnotation 1`] = `
"
export type Action = string

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow

export type Action = ;
"
`;

exports[`default no type Action: no type Action 1`] = `
"
// @flow

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow

export type Action = ;
"
`;

exports[`default payload: payload 1`] = `
"
export const INCREMENT: 'app/counter/Increment' = 'app/counter/Increment';

export type Increment = { type: typeof INCREMENT, payload: number };

export type Action = Increment;

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow

export type Increment = { type: typeof INCREMENT; payload: number; };

export type Action = Increment;
"
`;

exports[`default remove unuded action: remove unuded action 1`] = `
"
export const INCREMENT: 'app/counter/Increment' = 'app/counter/Increment';

export type Increment = { type: typeof INCREMENT, payload: number };

export type Action = Decrement;

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow

export type Decrement = {
  type: typeof constants.DECREMENT
};

export type Action = Decrement;
"
`;

exports[`default snake_case: snake_case 1`] = `
"
export type Action = ThisIsAction;

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow

export type ThisIsAction = {
  type: typeof constants.THIS_IS_ACTION
};

export type Action = ThisIsAction;
"
`;
